Samples/2.0/MathGrains/Client/Program.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Orleans;
+using Orleans.Runtime;
+using Orleans.Configuration;
+using GrainInterfaces;
+
+namespace OrleansClient
+{
+    /// <summary>
+    /// Orleans test silo client
+    /// </summary>
+    public class Program
+    {
+        static int Main(string[] args)
+        {
+            return RunMainAsync().Result;
+        }
+
+        private static async Task<int> RunMainAsync()
+        {
+            try
+            {
+                using (var client = await StartClientWithRetries())
+                {
+                    await DoClientWork(client);
+                    Console.ReadKey();
+                }
+
+                return 0;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                Console.ReadKey();
+                return 1;
+            }
+        }
+
+        private static async Task<IClusterClient> StartClientWithRetries(int initializeAttemptsBeforeFailing = 5)
+        {
+            int attempt = 0;
+            IClusterClient client;
+            while (true)
+            {
+                try
+                {
+                    client = new ClientBuilder()
+                        .UseLocalhostClustering()
+                        .Configure<ClusterOptions>(options =>
+                        {
+                            options.ClusterId = "dev";
+                            options.ServiceId = "HelloWorldApp";
+                        })
+                        .ConfigureLogging(logging => logging.AddConsole())
+                        .Build();
+
+                    await client.Connect();
```

Comment:
We should change this: some existing code does this, too, but there is a better way to make the client connection retry. We can use the `Connect(Func<Exception, Task<bool>>)` overload, like in #4711. The method will be much simpler as a result.

Reply:
made a similar cleanup to hello world.  When that get's pass review, you can model off of it, if you like.  Cleanup connect retry loop in hello world sample #4767